### PR TITLE
Add addtional SAP facts to system profile

### PR DESCRIPTION
**Pull request by CodeHeeler**
Pending schema PR: https://github.com/RedHatInsights/inventory-schemas/pull/25

### CODE REVIEW

**Review by kruai — APPROVED** [review]
LGTM, but would probably be safe to wait for an additional review from a veteran :)

**Comment by thearifismail** [comment]
Question... What's the source of`maxLength` values for `sap_sid`, `sap_instance_number`, and `sap_version_info`?  How do I know that the maxLengths in `system_profile_spec` are correct, i.e. is there no chance of `sap_instance_number` being 3 char long?

**Comment by CodeHeeler** [comment]
> @CodeHeeler, Question... What's the source of`maxLength` values for `sap_sid`, `sap_instance_number`, and `sap_version_info`? How do I know that the maxLengths in `system_profile_spec` are correct, i.e. is there no chance of `sap_instance_number` being 3 char long?

The source for the sap_instance_number is here:
https://help.sap.com/viewer/2c1988d620e04368aa4103bf26f17727/2.0.05/en-US/ae305ab18bad42efa7b067e7cc82b846.html

The source for the version number is an example from Bernd Finger in Red Hat and the example here as this is the output for --version:
https://answers.sap.com/questions/161894/hana-version-check.html

The source for the sid's being 3 is here:
https://help.sap.com/viewer/2c1988d620e04368aa4103bf26f17727/2.0.05/en-US/7672838aa9c44c7ab9d10188531669f3.html?q=sids

Anything longer is incorrect, so we can either make a longer length for cushion for incorrect values or we can validate here and correct reporters that are not using correct values, as with any of the other facts.

**Comment by thearifismail** [comment]
> Anything longer is incorrect, so we can either make a longer length for cushion for incorrect values or we can validate here and correct reporters that are not using correct values, as with any of the other facts.

No don't need to make longer as it has lower potential for bugs.  Was just curious because `maxLength` for `sap_sid` and `sap_instane_number` appeared too small.  If adjustments is needed in future, it can be dealt with then.

LGTM

**Review by thearifismail — APPROVED** [review]

**Review by ryandillinfelton — APPROVED** [review]

File: swagger/system_profile.spec.yaml [hunk 1]
```diff
@@ -245,6 +245,18 @@ $defs:
           example: "H2O"
           maxLength: 3
           pattern: '^[A-Z][A-Z0-9]{2}$'
+      sap_instance_number:
+        type: integer
+        description: The instance number of the SAP HANA system
+        example: '42'
+        maxLength: 2
+        pattern: '(?!99)(?!98)^[0-9]{2}$'
```
Comment (ryandillinfelton) [inline swagger/system_profile.spec.yaml:253]: Tahmid had a good improvement suggestion for this field https://github.com/RedHatInsights/inventory-schemas/pull/25/files#r509548114

File: app/models.py [hunk 2]
```diff
@@ -389,6 +389,15 @@ class SystemProfileSchema(MarshmallowSchema):
             validate=[marshmallow_validate.Length(max=3), marshmallow_validate.Regexp(regex=r"^[A-Z][A-Z0-9]{2}$")]
         )
     )
+    sap_instance_number = fields.Int(
+        validate=[marshmallow_validate.Length(max=2), marshmallow_validate.Regexp(regex=r"(?!99)(?!98)^[0-9]{2}$")]
```
Comment (ryandillinfelton) [inline app/models.py:393]: Using the [Range validation](https://www.programcreek.com/python/example/108225/marshmallow.validate.Range) method could make things cleaner here

**Review by ryandillinfelton — CHANGES_REQUESTED** [review]

**Review by ryandillinfelton — APPROVED** [review]

**Review by ryandillinfelton — APPROVED** [review]

**Review by tahmidefaz — APPROVED** [review]

**Pull request merged**